### ISSUE TITLE
Typescript this parameters don't work

### ISSUE THREAD

**Issue by ashrafsabrym** [swc-project/swc#236]
I'm trying to compile a function with [this parameters](https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters). Part of my code is:
```javascript
export function DirectiveIserviceIdPrefixController(this: any, $scope: angular.IScope, $attrs: angular.IAttributes)
{
  this.$scope = $scope;
  this.$attrs = $attrs;
}
```
And I get the error:
Module parse failed: Unexpected keyword 'this' (1:52)

Here's my Webpack configuration for swc:
```javascript
{
          test: /\.ts$/,
          exclude: [/node_modules/, /wwwroot/, /packages/],
          use: {
            loader: 'swc-loader',
            options: {
              jsc: {
                parser: {
                  syntax: 'typescript'
                }
              }
            }
          }
        }
```
The code compiles successfully with Babel, but with swc, I have to remove the `this` parameter and add the `noImplicitThis: false` configuration to tsconfig for my IDE to ignore it.

**Comment by kdy1** [comment]
I can't reproduce this.
See https://github.com/kdy1/swc/commit/08bbeff41dc120b8f0a39b57a91809304a051975 and https://github.com/kdy1/swc-example/tree/ed176bfe70252f997ea3fc5d52dc47f1ca6467e5/inputs/issue-236

What is your swc version?

**Comment by ashrafsabrym** [comment]
I certify that the issue occurs when using the `swc-loader` as part of Webpack, but the same code compiles successfully if swc-cli is used.

Please view [this repository](https://github.com/ashrafsabrym/test-swc). It isolates the issue.
Running `npm run build:dev` will cause the mentioned issue, while `npx swc ./directives/directive.ts -d ./` compiles the file successfully.

**Comment by kdy1** [comment]
I found the issue.
swc transcompiles it to 

```js
export function DirectiveController(this, $scope, $attrs) {
    this['$scope'] = $scope;
    this['$attrs'] = $attrs;
}
```

and webpack cannot understand this syntax.

So I guess swc should remove this parameter.

**Comment by ashrafsabrym** [comment]
Yes, I see.
That's why in [Typescript documentation](https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters) they call them _fake parameters_:
> this parameters are fake parameters that come first in the parameter list of a function

**Comment by kdy1** [comment]
As this is a bug fix, I'll publish 1.0.21 soon.

**Comment by kdy1** [comment]
Fixed with `@swc/core@1.0.21`
